src/core/packages/saved-objects/api-server-internal/src/lib/apis/update.ts

```diff
@@ -178,6 +182,13 @@ export const executeUpdate = async <T>(
 
   // UPSERT CASE START
   if (shouldPerformUpsert) {
+    // Note: Upsert does not support adding accessControl properties. To do so, the `create` API should be used.
+
+    if (registry.supportsAccessControl(type)) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        `Type "${type}" supports access control, but the "upsert" option does not support adding access control properties. Use the "create" API to create a new saved object with access control properties.`
```

Comment:
`update` doesn't allow changing `accessControl` during the non-upsert case either, so all we need to throw on is if the request is trying to explicitly set (or change) access control in an `update` operation.

```suggestion
        `"update" does not support setting access control or changing owner. Use `changeObjectAccessControl` instead.` 
```